src/schemas/server.ts

```diff
@@ -0,0 +1,154 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022-present Mirage Aegis
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import mongoose, { Schema, Document, InferSchemaType } from "mongoose";
+
+const serverSchema: Schema = new Schema({
+    _id: String,
+    logs: String,
+});
+
+type ServerT = InferSchemaType<typeof serverSchema>;
+
+/**
+ * The minimum time a server object stays in the cache.
+ * 20 minutes
+ */
+const serverLifetime: number = 1_200_000;
+
+export class Server {
+    /**
+     * The corresponding Mongo model used for reading and writing to the database.
+     */
+    private static readonly model = mongoose.model("Server", serverSchema);
+
+    /**
+     * The cached servers.
+     * The servers are stored along with the ID for the timer responsible for
+     * clearing the memory.
+     */
+    private static cache: Map<string, [Server, NodeJS.Timeout]> = new Map();
```

Comment:
The server configurations that are fetched from the database are cached to improve performance at the cost of memory